src/groovy/au/org/emii/portal/proxying/RequestProxyingController.groovy

```diff
@@ -44,10 +44,17 @@ abstract class RequestProxyingController extends HostVerifyingController {
                 return
             }
         }
-
         _addDownloadTokenCookie()
         _setDownloadFilename(response, params)
-        _makeRequest(request, response, params, paramProcessor, streamProcessor)
+
+        try {
+            _makeRequest(request, response, params, paramProcessor, streamProcessor)
+        }
+        catch (IOException e) {
+            def msg = String.format("Could not reach %s %s", probeUrl, e.message)
+            log.error msg
+            render text: msg, contentType: "text/html", encoding: "UTF-8", status: HTTP_500_INTERNAL_SERVER_ERROR
```

Comment:
I thought that the content type should be png

`Content-Type: image/png`

Reply:
Status should be a 504 Gateway Timeout: "The server was acting as a gateway or proxy and did not receive a timely response from the upstream server."

Reply:
I think contentType should be excluded, since we can't know what the proxied server would have returned if it was working.

Reply:
contentType is text/html as we are returning a message, but yes 504 Gateway timeout sounds better